### ISSUE TITLE
Need a list of the available downloadable connectors

### ISSUE THREAD

**Issue by Yashaswini1998** [infinyon/fluvio#4265]
Hi, can someone please direct me to the list of available downloadable list of inbound and outbound connectors?

The [hub link](https://www.fluvio.io/docs/hub/overview) only gives the details and not the commands to download the connectors using the cdk command as shown here for the HTTP inbound connector in the [quickstart](https://www.fluvio.io/docs/fluvio/quickstart/) "cdk hub download infinyon/http-source@0.3.8"

The config files given in the hub link ask for a complete RUST project to be built from scratch including files like Cargo.toml etc. Can someone please share a sample project I can refer to build my custom connectors as well?

**Comment by ajhunyady** [comment]
@digikata can you please share the details, and maybe the ‘hello world’ connector as well.

**Comment by Yashaswini1998** [comment]
@ajhunyady  Thank you so much for responding.

Just an update. I got the list of the available connectors by doing "fluvio hub connector list". However, the duckdb-sink connector says 404 error not found.

I'm trying the sql-sink connector now where the given config in the hub link uses json-sql smart module. The json-sql smart module also seems to be missing as this also throws a 404 Not Found error.
`$ cdk hub download infinyon/json-sql@0.2.1`
![Image](https://github.com/user-attachments/assets/5e8461de-95d7-4830-a9d1-46da743bfbac)

Also, when I try to deploy the connector using cdk deploy start command, it says connector started but after listing it says Stopped under status. 

I observed the same with the mqtt-source connector as well. Please find the screenshot below.
![Image](https://github.com/user-attachments/assets/4bc50f8c-888f-42c5-b1e2-51073363c8c9)

So, I think @digikata  and @ajhunyady it would be really helpful if you could share that "hello world" connector along with the step-by-step instructions to set it up.

**Comment by drc-infinyon** [comment]
The sql sink connector download command is 

`$ cdk hub list
  
  CONNECTOR                          Visibility 
  benmanns/http-source@0.3.10        public     
  day-os/http-sink-fork@0.1.5        public     
  infinyon-labs/graphite-sink@0.2.0  public     
  infinyon/duckdb-sink@0.1.0         public     
  infinyon/http-sink@0.2.11          public     
  infinyon/http-source@0.4.3         public     
  infinyon/ic-webhook-source@0.1.4   public     
  infinyon/inf-webhook-source@0.1.9  public     
  infinyon/kafka-sink@0.2.9          public     
  infinyon/kafka-source@0.2.7        public     
  infinyon/mqtt-source@0.2.9         public     
  infinyon/sql-sink@0.4.3            public     
  qdrant/qdrant-sink@0.1.0           public     
  sdf-test/conn-hello-source@0.2.0   public`

And then the download is:

`cdk hub download infinyon/sql-sink@0.4.3`

Can you share your connector config.yaml for the stopped connectors?

**Comment by ajhunyady** [comment]
@Yashaswini1998, smartmodules are downloaded on your cluster and they require a different command:

```
$ fluvio hub smartmodule download infinyon/json-sql@0.2.1
```

Unfortunately DuckDb connector was experimental and we did not update it. We'll move that out of the `infinyon` hub. 
However, http-sink should work as expected.

**Comment by Yashaswini1998** [comment]
@ajhunyady Okay, Thank you so much for your response. Can you please confirm if we can expect duckdb connector to be available anytime soon?

@drc-infinyon Thank you for your response too!
I'm using the config.yaml give in the hub documentation [here ](https://www.fluvio.io/docs/hub/connectors/inbound/mqtt) for the respective connectors. 
I'm following the below steps: (using mqtt-source example)
1) I download the connector `cdk hub download infinyon/mqtt-source@0.2.9`
2) Create the sample-config.yaml file with the given script in the link 
3) Deploy the connector `cdk deploy start --config sample-config.yaml --ipkg infinyon-mqtt-source-0.2.9.ipkg`
Says started connector
4) Immediately when I try to list the deployments it says stopped
![Image](https://github.com/user-attachments/assets/9ac9c0d2-fb62-4253-be62-1182fd078bfe)



Am I missing any step?

**Comment by ajhunyady** [comment]
what does `cdk deploy log` show?

**Comment by Yashaswini1998** [comment]
timed out searching metadata Partition failed due to timeout: 60000 ms

![Image](https://github.com/user-attachments/assets/171f1cd0-6a5e-4b3a-8de9-cbfd6d4a031d)

**Comment by ajhunyady** [comment]
A few questions:
* What environment are you running?
* Can you produce & consume from the CLI?
* What is the output of `fluvio partition list` ?

**Comment by Yashaswini1998** [comment]
I'm running on VM with
1. OS - Ubuntu 22.04.3 (64-bit)
2. RAM - 8 GB
3. Disk Space - 50 GB

Produce and consume from CLI was working earlier but is now throwing the same timed out error
![Image](https://github.com/user-attachments/assets/91c1df33-ccb0-461b-b465-1ad0b1f93d32)


This is the output of `fluvio partition list` but cluster says running
![Image](https://github.com/user-attachments/assets/43786aa5-403b-40e7-8d16-0d061e92dd11)

**Comment by ajhunyady** [comment]
There are some communication issues between the SC and the SPU.
Could you try deleting and recreating the cluster to see if that fixes the issue.

The connectors are essentially fluvio clients from the cluster's point of view.
If the cli produce/consume does not work, the connectors won't work either.

**Comment by Yashaswini1998** [comment]
The cluster works fine for sometime. But the observation is that I start getting these timed out errors after using it for a while. And won't recover even after deleting all the current deployments and redeploying connectors.

Anyway, for the time it's working, can you please share some example program showing a source connector with a smart module -> data transferred to sink -> sink connector with a smart module?

Basically, how do I build a complete pipeline with these connectors to send and receive data across devices?

This would be really helpful!

**Comment by sehz** [comment]
Hi @Yashaswini1998 ,  What's version of Fluvio and CPU Arch ( x86 or Arm64?).    There is log file in the `~/.fluvio/log.   Hop in our discord https://discord.com/invite/bBG2dTz, for more interactive discussion.
For more complex pipeline, would recommend to look at: https://www.fluvio.io/sdf/.   SDF can model complex dataflow and pipeline.   It's kind of next evolution of SmartModule.

**Comment by Yashaswini1998** [comment]
Hi @sehz, please find the details below
![Image](https://github.com/user-attachments/assets/2d214dcf-b279-4e8a-a507-5a65c9475da1)


I'm facing another problem, wherever I deploy and start the sql-sink connector, the log file grows rapidly and hogs up the disk space even if I'm not sending any data to the sink. There is nothing in the postgres SQL table as well to which the sql-sink connector is configured to write.
The size of log file for sql-sink connector reached 30.4 GB
![Image](https://github.com/user-attachments/assets/50676f9f-ae1f-454b-9bcf-00061c3f97a9)

The log file has the below messages
![Image](https://github.com/user-attachments/assets/e48410ac-59b8-405b-bc5b-fcbb958d9e4e)

**Comment by sehz** [comment]
That seems surprising.  Can you share your connector config? Did you set log level to DEBUG?

**Comment by Yashaswini1998** [comment]
Postgres version
`psql (PostgreSQL) 14.13 (Ubuntu 14.13-0ubuntu0.22.04.1)`

**Comment by sehz** [comment]
In order to diagnose this further, can you describe step by step instruction on how to reproduce problem or point to existing configuration?

**Comment by Yashaswini1998** [comment]
I'm following the steps given in https://www.fluvio.io/docs/hub/connectors/outbound/sql

1. Create the connector-config.yaml file using the sample with transformation given in the link.
2. Install postgres and create a database called mydatabase with the topic_message table using the CREATE table command given in the link
3. Replace the URL with my local postgres database url
4. Deploy the connector using the cdk deploy start command from the link
5. The connector is in the running state

Now, I leave it idle, no data pushed to the topic. Still the log file grows as big as 30+ GBs within 1-2 hrs.

**Comment by sehz** [comment]
I was able to get sql sink connecter to postgres running.  Can you share you configuration so could replicate it?
